invoke-ai/InvokeAI
# feat(ui): layers metadata recall

## Summary

Support metadata recall for layers.

TODO:

- [x] Do not store ephemeral layer state in metadata (e.g. 'uploadedMaskImage', maybe `bboxNeedsUpdate`)
- [x] Check images for existence? May as well implement this for all metadata recallers that use an image while I'm at it...
- [x] Decide how to handle metadata from layers into canvas
- [x] Decide whether to store both the old-style metadata and layers metadata - I don't think I want to store both...
- [x] Add versioning to metadata
- [x] Test thoroughly

## Related Issues / Discussions

n/a

## QA Instructions

When ready for review, use control layers and check the metadata tab. Try recalling an individual layer, or use all to get all layers.

## Merge Plan

n/a

## Checklist
